sql/src/main/java/org/apache/druid/sql/calcite/aggregation/DimensionExpression.java

```diff
@@ -28,6 +28,7 @@
 
 public class DimensionExpression
 {
+  private final String inputDimension;
```

Comment:
How about `inputColumn`? (I think it's a bit weird to call columns in the table "dimensions" and would like to get away from that in new code.)

It would also be good to include a comment about what the relationship is between `inputColumn` and `expression`, with some examples. It might get confusing to have both. Additionally: if the expression is a simple column reference, is it required that `expression.getDirectColumn().equals(inputColumn)`? If so it would be good to validate that in the constructor.

Reply:
Heh, I initially had it named `inputColumn` but changed to `inputDimension` to match the `DimensionSpec` which it creates.

However, since this field _actually_ is only used when creating a `DimensionSpec` when the `DimensionExpression` is for a virtual column, I have renamed the field to `virtualColumn`. When a `DimensionExpression` is a direct access or simple extraction, we use `expression.getSimpleExtraction().toDimensionSpec` to craft the `DimensionSpec` instead, so this field is ignored.

To help clarify this, i collapsed the constructors into a single private constructor and made static methods `DimensionExpression.ofSimpleColumn` and `DimensionExpression.ofVirtualColumn` for creating the two types of `DimensionExpression` we have, as per your other comment, which I think helps clear up usage.